### PR TITLE
Scoop.sh App Manifest

### PR DESCRIPTION
**Pull request by Southclaws**
Finished the Scoop.sh manifest generator for Windows releases!

See #387

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] I have read the **CONTRIBUTING** document.
* [x] `make ci` passes on my machine.
* [x] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: pipeline/scoop/scoop_test.go [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			for _, a := range tt.artifacts {
+				tt.args.ctx.Artifacts.Add(a)
+			}
+			err := doRun(tt.args.ctx, tt.args.client)
+			if tt.wantErr {
+				assert.Error(t, err)
```
Comment (caarlos0) [inline pipeline/scoop/scoop_test.go]: should be `tt` here

File: pipeline/scoop/scoop_test.go [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+			} else {
+				assert.NoError(t, err)
```
Comment (caarlos0) [inline pipeline/scoop/scoop_test.go]: should be `tt` here as well

File: pipeline/scoop/scoop_test.go [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+				Description: "A run pipe test formula",
+			},
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotResult, err := buildManifest(tt.args.ctx, tt.args.client, tt.args.artifacts)
+			if tt.wantErr {
+				assert.Error(t, err)
```
Comment (caarlos0) [inline pipeline/scoop/scoop_test.go]: in this method as well, should be `tt` instead of `t`

File: pipeline/scoop/scoop_test.go [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+		{
+			"custom",
+			args{&context.Context{Version: "1.0.0", Config: config.Project{Release: config.Release{GitHub: config.Repo{Owner: "user", Name: "repo"}}}}, "https://git.my.company.com", "file.tar.gz"},
+			"https://git.my.company.com/user/repo/releases/download/1.0.0/file.tar.gz",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotURL := getDownloadURL(tt.args.ctx, tt.args.githubURL, tt.args.file)
+			assert.Equal(t, tt.wantURL, gotURL)
```
Comment (caarlos0) [inline pipeline/scoop/scoop_test.go]: here too

File: pipeline/scoop/scoop.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+// Resource represents a combination of a url and a binary name for an architecture
+type Resource struct {
+	URL string `json:"url"` // URL to the archive
+	Bin string `json:"bin"` // name of binary inside the archive
+}
+
+func buildManifest(ctx *context.Context, client client.Client, artifacts []artifact.Artifact) (result bytes.Buffer, err error) {
+	var githubURL = "https://github.com"
+	if ctx.Config.GitHubURLs.Download != "" {
+		githubURL = ctx.Config.GitHubURLs.Download
```
Comment (caarlos0) [inline pipeline/scoop/scoop.go]: can you add a TODO here to make this default in some defaulter? (it's probably like this in more places, so let me fix it in another PR 👍 )

File: pipeline/scoop/scoop_test.go (hunk 1 above)
Comment (Southclaws) [reply pipeline/scoop/scoop_test.go]: `tt` is a struct, not a `*testing.T`. `t` is shadowed by the inline function, the test generator does this.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 1e0ff86d14f20bfdee662c091f428d4f6e976e0f
  * Commits must be in the following format: **%{type}: %{description}**

Guidelines are available at https://github.com/goreleaser/goreleaser/blob/master/CONTRIBUTING.md#create-a-commit

---
This message was auto-generated by https://gitcop.com

File: pipeline/scoop/scoop_test.go (hunk 1 above)
Comment (caarlos0) [reply pipeline/scoop/scoop_test.go]: oh, my bad, read it wrong, sorry :D

**Comment by caarlos0** [comment]
Nice!

I think the only thing missing now is the docs, can you add that?

Also: what do I need to do to publish goreleaser to scoop? We should probably document how to install goreleaser with scoop as well in that case =D

**Comment by Southclaws** [comment]
Yep, I can work on some docs tonight! I'm still trying to get a few error branches to trigger in coverage. They are actually hit during tests (I tested via step-through debugging) but for some reason, coverage marks the branches as not hit... I'll keep going at it though.

**Comment by caarlos0** [comment]
repo created -> https://github.com/goreleaser/scoop-bucket :D

**Comment by Southclaws** [comment]
On the topic of unit tests, I noticed that codecov actually reports correct coverage and it all seems 100% fine - the Go vscode plugin seem to be the problem on my end. The only issue is this line https://codecov.io/gh/goreleaser/goreleaser/src/a40353674f6abd1d84e9d825bc4f7d4b656691c5/pipeline/scoop/scoop.go#L133 which seems impossible to trigger given that `Manifest` is entirely made of valid JSON types, there seems no way to force an error case here. So it seems the only way to keep codecov happy for this pull would be to adjust the requirements or modify `Run` to support a dummy client (which would require work across the entire codebase).

But other than that, if you're happy with the docs, this looks ready to merge!

**Comment by caarlos0** [comment]
LGTM

**Pull request merged**